RecoTauTag/HLTProducers/src/L1TCaloJetsMatching.cc

```diff
@@ -0,0 +1,113 @@
+#include "RecoTauTag/HLTProducers/interface/L1TCaloJetsMatching.h"
+#include "Math/GenVector/VectorUtil.h"
+#include "DataFormats/HLTReco/interface/TriggerTypeDefs.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "DataFormats/JetReco/interface/CaloJet.h"
+
+//
+// function that splits a jet collection into two 
+//
+std::pair<reco::CaloJetCollection,reco::CaloJetCollection> categorise(const reco::CaloJetCollection& matchedJets, double pt1, double pt2, double m_jj)
+{
+  std::pair<reco::CaloJetCollection,reco::CaloJetCollection> output;
+  
+  unsigned int iMax = 0;
+  unsigned int jMax = 0;
+  double mjjMax = 0;
+
+  if (matchedJets.size() > 1){
+    for (unsigned int i = 0; i < matchedJets.size()-1; i++){
+      for (unsigned int j = i+1; j < matchedJets.size(); j++){
+
+        const reco::CaloJet &  myJet1 = (matchedJets)[i];
+        const reco::CaloJet &  myJet2 = (matchedJets)[j];
+        
+        double mjj = (myJet1.p4()+myJet2.p4()).M();
+        if (mjj > mjjMax){
+          mjjMax = mjj;
+          iMax = i;
+          jMax = j;
+        }
+      }
+    } 
+    const reco::CaloJet &  myJet1 = (matchedJets)[iMax];
+    const reco::CaloJet &  myJet2 = (matchedJets)[jMax];
+        
+    if ((myJet1.pt() > pt1) && (myJet2.pt() > pt2) && (mjjMax > m_jj)){
+      output.first.push_back(myJet1);
+      output.first.push_back(myJet2);
+    }
+    
+    else if ((myJet1.pt() < pt1) && (myJet1.pt() > pt2) && (myJet2.pt() > pt2) && (mjjMax > m_jj)){
+      const reco::CaloJet &  myJet3 = (matchedJets)[0];
+      if (myJet3.pt()>pt1){
+        output.second.push_back(myJet1);
+        output.second.push_back(myJet2);
+        output.second.push_back(myJet3);
+      }
+    }    
+  }
+  return output;    
+}
+
+//
+// class declaration
+//
+L1TCaloJetsMatching::L1TCaloJetsMatching(const edm::ParameterSet& iConfig):
+  jetSrc_    ( consumes<reco::CaloJetCollection>             (iConfig.getParameter<edm::InputTag>("JetSrc"      ) ) ),
+  jetTrigger_( consumes<trigger::TriggerFilterObjectWithRefs>(iConfig.getParameter<edm::InputTag>("L1JetTrigger") ) ),
+  mPt1_Min   ( iConfig.getParameter<double>("Pt1Min")),
+  mPt2_Min   ( iConfig.getParameter<double>("Pt2Min")),
+  mMjj_Min   ( iConfig.getParameter<double>("MjjMin"))
+{  
+  produces<reco::CaloJetCollection>("TwoJets");
+  produces<reco::CaloJetCollection>("ThreeJets");
+}
+L1TCaloJetsMatching::~L1TCaloJetsMatching(){ }
+
+void L1TCaloJetsMatching::produce(edm::StreamID iSId, edm::Event& iEvent, const edm::EventSetup& iES) const
+{    
+  std::unique_ptr<reco::CaloJetCollection> caloMatchedJets(new reco::CaloJetCollection);
```

Comment:
actually, you do not put this in the Event, so you should simply use a local variable
```
reco::CaloJetCollection caloMatchedJets;
```